src/condor_tests/test_multifile_curl_plugin_timeout.py

```diff
@@ -0,0 +1,207 @@
+    contents = f.readlines()
+    f.close()
+    return contents
+
+#--------------------------------------------------------------------------
+# Unset HTTP_PROXY for correct operation in Docker containers
+lowered = dict()
+for k in os.environ:
+    lowered[k.lower()] = k
+os.environ.pop(lowered.get("http_proxy", "http_proxy"), None)
+
+#--------------------------------------------------------------------------
+#Generic strings of a specific data size
+ONE_BYTE_STR   = 'n'
+TEN_BYTE_STR   = 'n' * 10
+HUND_BYTE_STR  = 'n' * 100
+ONE_KB_STR     = 'n' * 1024
+
+#Transfer rate test cases
+#   -Transfer goes in order of top down
+#   -To add a delay add "Delay={Delay length}"
+#    Note: Did this rather than | lambda: str(sleep(num))
+#          because it would sleep but also yield 'None'
+#          thus adding 4 extra bytes of unwanted data transfer
+#   -For bytes sent either use one of the generic strings above
+#    or do | lambda: 'char' * num_bytes
+
+#!!!!!!!Note that the plugin checks for >1byte/sec ~every 30 seconds!!!!!!!
+TEST_CASES = {
+    #Successful xfer test cases
+    "pass_trailing_data": (
+        #Massive xfer, delay, small xfer, done (1024:30,1:fin)
+        lambda: ONE_KB_STR,
+        lambda: "Delay=30",
+        lambda: ONE_BYTE_STR,
+    ),
+    "pass_normal_end": (
+        #Massive xfer, delay, enough xfer, delay, done (1024:30,35:30)
+        lambda: ONE_KB_STR,
+        lambda: "Delay=30",
+        lambda: 'n' * 35,
+        lambda: "Delay=30",
+    ),
+    #Failure xfer test cases
+    "fail_slow_middle": (
+        #Massive xfer, very long delay, timeout (1024:30,0:30)
+        lambda: ONE_KB_STR,
+        lambda: "Delay=60",
+        lambda: ONE_BYTE_STR,
+    ),
+    "fail_delayed_start": (
+        #timeout (0:30)
+        lambda: "Delay=30",
+        lambda: TEN_BYTE_STR,
+    ),
+    "fail_slow_start": (
+        #Not enough xfer, timeout (20:30)
+        lambda: 'n' * 20,
+        lambda: "Delay=30",
+        lambda: ONE_BYTE_STR,
+    ),
+}
+
+#--------------------------------------------------------------------------
+#Internal Iterator class to send xfer data correctly
+class SlowIterator:
+    def __iter__(self):
+        for data in (l() for l in self.res):
+            #print(data)
+            if "Delay" in data:
+                time = data.split("=")
+                sleep(int(time[1]))
+                continue
+            else:
+                yield data
+    def __init__(self,response):
+        self.res = response
+
+#Handler fuction to return response of iterator object data
+def req_handler(request, response):
+    return Response(SlowIterator(response))
+
+TEST_NUM = -1
+#Server handling of an http request
+def server_data_xfer(srv,response,key):
+    global TEST_NUM
+    TEST_NUM += 1
+    srv.expect_request(f"/testserver/{TEST_NUM}/{key}").respond_with_handler(lambda r: req_handler(r,response))
+    return "http://localhost:{0}/testserver/{1}/{2}".format(srv.port,TEST_NUM,key)
```

Comment:
We discussed this on Slack, and I think it would be good to remove `TEST_NUM`; it's unnecessary and therefor confusing.